python/ray/data/_internal/logging.py

```diff
@@ -186,15 +193,37 @@ def configure_logging() -> None:
 
     If "RAY_DATA_LOG_ENCODING" is specified as "JSON" we will enable JSON logging mode
     if using the default logging config.
+
+    Note: This function is idempotent after initial configuration when handlers have been
+    added to ray.data loggers, as reconfiguration would close and invalidate those handlers.
     """
+    global _configured_logger_handlers
+    global _logging_configured
 
-    # Dynamically load env vars
+    # Dynamically load env vars and get config
     config_path = os.environ.get(RAY_DATA_LOGGING_CONFIG_ENV_VAR_NAME)
     log_encoding = os.environ.get(RAY_DATA_LOG_ENCODING_ENV_VAR_NAME)
     config = _get_logging_config()
 
+    # Check if any managed loggers already have handlers before configuring
+    # If so, skip configuration to preserve them since dictConfig() would close them
+    configured_logger_names = set(config.get("loggers", {}).keys())
+    for name, logger_obj in logging.root.manager.loggerDict.items():
```

Comment:
Why not checking for Ray Data root loggers only (ie `ray.data.*`, `ray.anyscale.data.*`)